docs/concepts/runtime_capabilities.md

```diff
@@ -103,21 +103,85 @@ Both the [`elastic` connector](../reference/connectors/elastic.md) and the [`htt
+
+```mermaid
+graph LR
+    A{kafka_consumer} -->|send| B(pipeline)
+    B -->|send| C{udp_client}
+```
+
+**Source**: The `kafka_consumer` connector, when used with `transactional` mode supports guaranteed delivery. A failure on a message after the latest acknowledged offset will revert to that message and replay all data since then.
+
+**Sink**: The `udp_client` connector offers no delivery guarantee. Any message reaching it will trigger an acknowledgment automatically.
+
+**Result**: All messages read from Kafa will be automatically acknowledged.
+
+### Example 2: from udp_server to kafka_producer
+
+```mermaid
+graph LR
+    A{udp_server} -->|send| B(pipeline)
+    B -->|send| C{kafka_producer}
+```
+
+**Source**: UDP as transport protocol offers no guarantees or order, the upstream system sending the messages does not know if it arrived or is handled. Thus the `udp_server` simply cannot make any guarantees, or mark any packets as delivered or not.
+
+**Sink**: Depending on its setting, the `kafka_producer` connector acknowledges successful production of the event and will fail an event if it can't produce it to the configured topic.
+
+**Result**: The upstream system has no guarantees if its messages arrived anywhere. If a message is delivered to tremor it will make a best effort attempt to deliver it, but if Kafka fails to persist the event, a replay can't happen due to UDP's limitations.
+
+### Example 3: from http_server to kafka_producer
+
+```mermaid
+graph LR
+    A{http_server} -->|send| B(pipeline)
+    B -->|send| C{kafka_producer}
+```
+
+**Source**: The `http_server` connector will acknowledge successful delivery of the message to its upstream system but does not support replays. Events that did fail inside tremor trigger a `500 Internal Server Error` HTTP response.
+
+**Sink**: Depending on its setting, `kafka_producer` acknowledges successful storage of the message and will fail an event if it can't store it.
+
+**Result**: Events that are delivered to tremor will be delivered to Kafka. If the `kafka_producer` fails to persist the message, an error is sent back to the upstream system as a 500 response to the HTTP request. How the upstream system handles this failure is up to the service.
+
+### Example 4: from http_server via wal connector to kafka_producer
+
+```mermaid
+graph LR
+    A{http_server} -->|send| B(pipeline http_wal)
+    B -->|send| C{wal}
+    C -> |send| D(pipeline wal_kafka)
+    D -->|send| E{kafka_producer}
+```
+
+**Source**: The `http_server` connector will acknowledge successful delivery of the message to its upstream system but does not support replays. Events that did fail inside tremor trigger a `500 Internal Server Error` HTTP response.
+
+**Intermidiary**: This setup uses a [`wal` connector](../connectors/wal.md) as an intermediary, effectively splitting the system into two pipelines. One pipeline being `http_server` to `wal` the other being `wal` to `kafka_producer`.
```

Comment:
My bad!

```suggestion
**Intermidiary**: This setup uses a [`wal` connector](../reference/connectors/wal.md) as an intermediary, effectively splitting the system into two pipelines. One pipeline being `http_server` to `wal` the other being `wal` to `kafka_producer`.
```

Reply:
hah you were quicker I was just tracking this down :D